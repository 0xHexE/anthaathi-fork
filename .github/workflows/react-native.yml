name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
