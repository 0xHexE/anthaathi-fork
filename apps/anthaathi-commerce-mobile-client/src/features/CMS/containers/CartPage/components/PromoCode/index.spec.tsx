import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import PromoCode from './index';

describe('PromoCode', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PromoCode title="Add your promo code" />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('promoCode')).toBeTruthy();
  });

  it('should call when we call tap handlePress', function () {
    const onpress = jest.fn();

    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PromoCode title="Add your promo code" handlePress={onpress} />
        </IntlProvider>
      </ThemeProvider>,
    );

    fireEvent.press(temp.queryByTestId('handlePress')!);
    expect(onpress).toBeCalledTimes(1);
  });

  it('should have PromoCode title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PromoCode title="Add your promo code" />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Add your promo code')).toBeTruthy();
  });
});
