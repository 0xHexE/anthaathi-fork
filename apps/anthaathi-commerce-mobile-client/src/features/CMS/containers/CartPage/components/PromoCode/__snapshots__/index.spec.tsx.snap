// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoCode should render the item 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginHorizontal": 10,
      "marginVertical": 5,
    }
  }
  testID="promoCode"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(234, 221, 255, 1)",
        "borderRadius": 4,
        "height": 48,
        "justifyContent": "center",
        "width": "100%",
      }
    }
    testID="handlePress"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(28, 27, 31, 1)",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#737272",
              "fontSize": 14,
              "fontWeight": "500",
            },
          ]
        }
      >
        Add your promo code
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "rgba(103, 80, 164, 1)",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï€«
      </Text>
    </View>
  </View>
</View>
`;
